### ISSUE TITLE
Dropped sms

### ISSUE THREAD

**Issue by expez** [WhisperSystems/Signal-Android#4784]
Signal was just updated, so this bug report is for 3.6.0.

[Debug log](https://gist.github.com/13c91babb99171cc5ce9), unfortunately after the automatic software update.

Every month, on the 1st, my power company sends me an sms asking for the current value of the power meter in my apartment, to keep track of usage.  I never received their sms yesterday.

I know it was sent, because I also have an app installed which looks up unknown phone numbers and this app triggered and displayed a notification that it had looked up `+47 2262`, which is the expected number.

I'm aware of two other dropped sms.  I figured the first one was a network failure, and told my gf to turn on delivery notification.  After doing that, she told me one of her messages went undelivered the other day.

**Comment by spank94** [comment]
+1 Same for me

**Comment by Ryan6578** [comment]
I'm having this issue too, only I have more problems than just that. A phone reboot seems to fix the issue of not receiving SMS messages (these are all unsecure messages - ie. not sent through Signal). But, I have people that send me messages I don't receive, and they either trickle in later on, or if I restart my phone, they come flooding in. I also have an issue with not being able to send messages. I have a network and WiFi signal, but when I try sending the message, it continuously fails.

Carrier: Verizon
Phone: HTC One M8
Android Version: 5.1
Signal Version: 3.6.1

**Comment by expez** [comment]
> A phone reboot seems to fix the issue of not receiving SMS messages (these are all unsecure messages - ie. not sent through Signal). 

A reboot did nothing for me.  The dropped messages were unsecure in my case too.

**Comment by moxie0** [comment]
@expez Please post a debug log immediately after a message is dropped.

**Comment by expez** [comment]
I switched back to Signal as the default messaging app today and got a few people to send me messages, but I couldn't reproduce the problem.

Being able to reliably send and receive messages is the most important feature of a messaging app to me, so I'm just gonna switch back and not spend any more time on this.

**Comment by davepwsmith** [comment]
See also: #4851

This should be reopened as a priority, surely!

**Comment by jarlelin** [comment]
I'll add that my problem is with unsecured messages (sms) as well.
Sort of hard to reproduce given its random nature.

**Comment by moxie0** [comment]
@davepwsmith As soon as someone posts a debug log we can take action. There's nothing we can do without one.

**Comment by spank94** [comment]
One droped sms today and here's my debug log => https://gist.github.com/fe10e940efdf7b12020e

**Comment by robryk** [comment]
I also had an SMS message dropped.

What happened:
1. I heard a notification sound.
2. I unlocked my phone, opened a notification about a new SMS message.
3. I read the message (the message was shown to have come from +480800800800 phone number for some weird reason; it was a message from my mobile phone company).
4. I opened the notification shade and tapped on another notification from Signal.
5. An empty conversation with "X also uses Signal" has appeared (the phone number of the contact in this conversation ends with 39).
6. I couldn't find the SMS message anywhere.
7. I tried closing (ie. swiping sideways in the running apps carousel -- this stops the activity IIRC) the Signal all and opening it again to no avail.
8. I took a debug log: https://gist.github.com/anonymous/239e55b13f2583826b018b096c136749

I did check the system SMS app, in case I managed to switch the default back without noticing. It also wasn't there.
